custom_components/solaredge_modbus_multi/switch.py

```diff
@@ -176,3 +180,52 @@ async def async_turn_off(self, **kwargs):
         _LOGGER.debug(f"set {self.unique_id} bits {set_bits:016b}")
         await self._platform.write_registers(address=57344, payload=set_bits)
         await self.async_update()
+
+
+class SolarEdgeGridControl(SolarEdgeSwitchBase):
+    entity_category = EntityCategory.CONFIG
+
+    def __init__(self, platform, config_entry, coordinator):
+        super().__init__(platform, config_entry, coordinator)
+        """Initialize the sensor."""
+
+    @property
+    def available(self) -> bool:
+        return super().available and self._platform.advanced_power_control
+
+    @property
+    def unique_id(self) -> str:
+        return f"{self._platform.uid_base}_grid_control"
+
+    @property
+    def name(self) -> str:
+        return "Grid Control"
+
+    @property
+    def is_on(self) -> bool | None:
+        try:
+            if self._platform.decoded_model["I_AdvPwrCtrlEn"] == 0x1:
```

Comment:
The return of the comparison is already a bool that you can return directly, no need for the `if .. else`
```suggestion
            return self._platform.decoded_model["I_AdvPwrCtrlEn"] == 1
```